addons/hr_timesheet/views/project_views.xml

```diff
@@ -40,7 +40,7 @@
             <field name="priority">24</field>
             <field name="arch" type="xml">
                 <button name="toggle_active" position="before">
-                    <button class="oe_stat_button" name="%(act_hr_timesheet_line_by_project)d" type="action" icon="fa-calendar" string="Timesheets" attrs="{'invisible': [('allow_timesheets', '=', False)]}" groups="hr_timesheet.group_hr_timesheet_user"/>
+                    <button class="oe_stat_button" name="%(act_hr_timesheet_line_by_project)d" type="action" icon="fa-calendar" string="Timesheets" attrs="{'invisible': ['|', ('allow_timesheets', '=', False), ('task_count', '=', 0)]}" groups="hr_timesheet.group_hr_timesheet_user"/>
```

Comment:
I do not agree with this change : you can have timesheet on project but not linked to tasks. Removing this button will make them inaccessible.